src/Service/Controllers/RestController.cs

```diff
@@ -185,8 +191,25 @@ private async Task<IActionResult> HandleOperation(
             string route,
             EntityActionOperation operationType)
         {
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            // This activity tracks the entire REST request.
+            using Activity? activity = TelemetryTracesHelper.DABActivitySource.StartActivity($"{HttpContext.Request.Method} {(route.Split('/').Length > 1 ? route.Split('/')[1] : "")}");
+            if (activity is not null)
+            {
+                activity.TrackRestControllerActivityStarted(
+                    HttpContext.Request.Method,
+                    HttpContext.Request.Headers["User-Agent"].ToString(),
+                    operationType.ToString(),
+                    route,
+                    HttpContext.Request.QueryString.ToString(),
+                    HttpContext.User.FindFirst("role")?.Value,
```

Comment:
We also check for the value of the request header `X-MS-API-ROLE` (if set) to determine what the role the request should be executed as.. instead of finding the first using `FindFirst`. 
Can we use the value of this header here and default to `FindFirst` if that doesnt exist?